### PR TITLE
Improve error handling in CI test launcher

### PR DESCRIPTION
**Pull request by elliot-nelson**
## Description
Small tweaks to the `ci.js` test launcher:

- Ensure that if an error occurs in the launcher (mismatched Chrome browser + ChromeDriver version, or missing GeckoDriver, etc.), the details of the error will be logged to console.
- Ensure that if the driver fails to start, the server still gets closed down, allowing the test launcher to exit immediately instead of hanging forever.
- As a side benefit, the `ci.js` launcher now works in latest node v8.x _and_ v10.x.  (Although I'm fine bringing the `.finally` back if this isn't important -- we only have ~6 months to live on node 8.)

## Motivation and Context
Minor improvements that make it easier for new contributors to run CI tests in the browser locally.  Probably has no effect on CI (although for corner cases, may improve debugging).

## How Has This Been Tested?
- So far, tested locally only, with latest Safari, Chrome, and Firefox.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: ci.js [hunk 1]
```diff
@@ -120,6 +120,17 @@ async function getResults(driver) {
   return {specResults: flatten(specResults), failedSuiteResults: flatten(failedSuiteResults)};
 }
 
+function cleanup() {
+  return Promise.resolve().then(() => {
```
Comment (slackersoft) [inline ci.js]: What's the reason for this wrapping `Promise.resolve().then`? I would expect just `Promise.all().catch` to work correctly here.

Comment (elliot-nelson) [reply ci.js]: We could probably scrap that, actually...

Originally I couldn't tell what was going wrong, because the error message was getting lost in the ether, so I wrapped the whole thing.  Then afterwards I realized the issue was driver wasn't defined yet, and added the inline checks.

TLDR: both checks are too much and could be simplified now.

Comment (slackersoft) [reply ci.js]: Sweet. Please update to include the simplified checks. Thanks!

**Comment by elliot-nelson** [comment]
Simplified check (removed extra promise wrapper).
